From fb6e495ba77fa15ec42dca29067c3158c6447db4 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 18 Dec 2019 16:50:33 +0200
Subject: [PATCH] [SW-0] pll: remove no more used w/a code

We do not have anymore old Gaudi SoCs, only the "fixed" ones remain.
Therefore we no longer need the PLL clock switch glitch W/A.
Remove the unused PLL clock switch W/A code.

Change-Id: Ifc0be7b8ddcb3d6538bd3359bc4854ee55934ac6
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/preboot.c b/app/preboot.c
index d51d909..d8c409d 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -147,7 +147,6 @@
 	hl_uart_init();
 	preboot_version();
 	hl_pll_init();
-	hl_pll_wa();
 	/* reinit uart after PLLs configured */
 	hl_uart_init();
 	preboot_copy_spl();
diff --git a/common/pll.c b/common/pll.c
index bad1bb3..6a298eb 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -103,15 +103,6 @@
 
 	for (sel = 0; sel < HL_PLL_DIV_SIZE; sel++) {
 		reg = cfg->base + HL_PLL_DIV_SEL + (sel * sizeof(u32));
-		switch (cfg->div_wa[sel]) {
-		case HL_PLL_WA_INIT:
-		case HL_PLL_WA_NOGT:
-			continue;
-		case HL_PLL_WA_NOWA:
-		default:
-			break;
-		}
-
 		switch (cfg->div_sel[sel]) {
 		case HL_PLL_DIV_REF:
 			regmap_writel(HL_PLL_DIV_REF, reg);
diff --git a/common/pll.h b/common/pll.h
index 3524ae9..d53f382 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -40,12 +40,6 @@
 	HL_PLL_DIV_PLL,
 };
 
-enum hl_pll_wa_sel {
-	HL_PLL_WA_NOWA = 0,
-	HL_PLL_WA_INIT,
-	HL_PLL_WA_NOGT,
-};
-
 enum hl_pll_div_out {
 	HL_PLL_DIV0,
 	HL_PLL_DIV1,
@@ -63,7 +57,6 @@
 	u32	nb;
 	int	div_factor[HL_PLL_DIV_SIZE];
 	int	div_sel[HL_PLL_DIV_SIZE];
-	int	div_wa[HL_PLL_DIV_SIZE];
 };
 
 enum hl_pll_bypass_val hl_pll_bypass_get(u32 pll_base,
@@ -74,6 +67,5 @@
 u32 hl_pll_get_freq(u32 pll_base, enum hl_pll_div_out div);
 bool hl_pll_is_configurable(struct hl_pll_cfg *cfg);
 void hl_pll_udelay(unsigned int usec);
-void hl_pll_wa(void);
 
 #endif /* _HL_PLL_H_ */
diff --git a/gaudi/asic.c b/gaudi/asic.c
index 1ba764b..7c0fe3b 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -26,11 +26,6 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "IF_W", .base = mmIF_W_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
@@ -39,11 +34,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
-		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
 		},
 	}, {
 		.name = "IF_E", .base = mmIF_E_PLL_BASE,
@@ -54,11 +44,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "DMA_E", .base = mmDMA_E_PLL_BASE,
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
@@ -67,11 +52,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
-		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
 		},
 	}, {
 		.name = "MESH_W", .base = mmMESH_W_PLL_BASE,
@@ -82,11 +62,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "SRAM_W", .base = mmSRAM_W_PLL_BASE,
 		.nr = 0, .nf = 35, .od = 1, .nb = 17, /* 900 */
@@ -95,11 +70,6 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
-		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
 		},
 	}, {
 		.name = "MESH_E", .base = mmMESH_E_PLL_BASE,
@@ -110,11 +80,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "SRAM_E", .base = mmSRAM_E_PLL_BASE,
 		.nr = 0, .nf = 35, .od = 1, .nb = 17, /* 900 */
@@ -124,11 +89,6 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "MME", .base = mmPSOC_MME_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
@@ -137,11 +97,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
-		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
 		},
 	}, {
 		.name = "PCI", .base = mmPSOC_PCI_PLL_BASE,
@@ -153,11 +108,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "TPC", .base = mmPSOC_TPC_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
@@ -166,11 +116,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
-		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
 		},
 	}, {
 		.name = "HBM", .base = mmPSOC_HBM_PLL_BASE,
@@ -181,11 +126,6 @@
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "NIC0", .base = mmNIC0_PLL_BASE,
 		.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
@@ -194,11 +134,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK
-		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
 		},
 	}, {
 		.name = "NIC1", .base = mmNIC1_PLL_BASE,
@@ -209,11 +144,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "NIC2_HBM", .base = mmNIC2_HBM_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 3, .nb = 31, /* 800 MHz */
@@ -222,11 +152,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
-		},
-		.div_wa = { HL_PLL_WA_NOWA,	/* Can't gate NIC2_HBM */
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
 		},
 	}, {
 		.name = "NIC2_MME", .base = mmNIC2_MME_PLL_BASE,
@@ -237,11 +162,6 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	}, {
 		.name = "NIC2_TPC", .base = mmNIC2_TPC_PLL_BASE,
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
@@ -251,35 +171,14 @@
 			     HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL
 		},
-		.div_wa = { HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA
-		},
 	},
 };
-
-static bool gaudi_hbw_init_done(u64 pll_base)
-{
-	return hl_pll_bypass_get(pll_base, HL_PLL_DIV0) == HL_PLL_PLL_CLK;
-}
-
-static bool gaudi_reset_soc_done(void)
-{
-	return regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2) == 1;
-}
 
 bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
 {
 	switch (cfg->base) {
 	case mmPSOC_CPU_PLL_BASE:
 		return false;
-	case mmPSOC_PCI_PLL_BASE:
-	case mmNIC2_HBM_PLL_BASE:
-		break;
-		if (gaudi_hbw_init_done(cfg->base) && gaudi_reset_soc_done())
-			return false;
-		break;
 	default:
 		break;
 	}
@@ -324,112 +223,4 @@
 	factor = pcie_div2_freq / HL_SPI_FLASH_MAX_FREQ;
 
 	return factor > 2 ? factor : 2;
-}
-
-static void gaudi_reset_soc(void)
-{
-	/* move the UART/SPI clock to reference clock */
-	hl_pll_udelay(10);
-	regmap_writel(HL_PLL_REF_CLK, mmPSOC_PCI_PLL_DIV_SEL_2);
-	hl_pll_udelay(1);
-	hl_uart_init();
-
-	regmap_writel(0xFFE00000, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_L);
-	regmap_writel(0x003FFFFF, mmPSOC_GLOBAL_CONF_SW_ALL_RST_CFG_H);
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_SW_ALL_RST);
-
-	/* Wait for reset */
-	__asm__ __volatile__ ("wfe\n\t" : : );
-
-	while (1)
-		preboot_puts("Surprise! Woke up from WFE...\n");
-}
-
-static u32 hl_pll_build_gate_mask(struct hl_pll_cfg *cfg)
-{
-	int i, gmask = 0;
-
-	for (i = 0; i < HL_PLL_DIV_SIZE; i++)
-		if (cfg->div_wa[i] == HL_PLL_WA_INIT)
-			gmask |= 1 << i;
-
-	return ~gmask & 0xF;
-}
-
-static void hl_pll_wa_clk_sw(struct hl_pll_cfg *cfg)
-{
-	int i;
-	u32 gmask = hl_pll_build_gate_mask(cfg);
-	u32 reg;
-
-	if (gmask != 0xF) {
-		regmap_writel(gmask, cfg->base + HL_PLL_CLK_GATER);
-	} else {
-		preboot_puts_raw("PLL ");
-		preboot_puts_raw(cfg->name);
-		preboot_puts(" Gate mask is 0xF - skip gating...\n");
-	}
-	hl_pll_udelay(1);
-
-	for (i = 0; i < HL_PLL_DIV_SIZE; i++) {
-		if (cfg->div_wa[i] == HL_PLL_WA_NOWA)
-			continue;
-
-		reg = cfg->base + HL_PLL_DIV_SEL + (i * sizeof(u32));
-		/* if DIV_PLL - first change DIV_REF, then DIV_PLL */
-		if (cfg->div_sel[i] == HL_PLL_DIV_PLL) {
-			regmap_writel(HL_PLL_DIV_REF, reg);
-			hl_pll_udelay(1);
-		}
-
-		regmap_writel(cfg->div_sel[i], reg);
-	}
-	hl_pll_udelay(10);
-
-	if (gmask != 0xF) {
-		regmap_writel(0xF, cfg->base + HL_PLL_CLK_GATER);
-		hl_pll_udelay(10);
-	}
-}
-
-void hl_pll_wa(void)
-{
-	int i;
-	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
-
-	/* H/W bug WA:
-	 * rdata[31:0] = strap_read_val;
-	 * wdata[31:0] = rdata[30:21],1'b0,rdata[20:0]
-	 */
-	boot_strap =	((boot_strap & 0x7FE00000) << 1) |
-			( boot_strap & 0x001FFFFF);
-
-	for (i = 0; i < ARRAY_SIZE(gaudi_pll_cfg); i++) {
-		switch (gaudi_pll_cfg[i].base) {
-		case mmPSOC_CPU_PLL_BASE:
-			continue;
-		case mmNIC2_HBM_PLL_BASE:
-			continue;
-			if (gaudi_reset_soc_done()) {
-				regmap_writel(0, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-				/* reinitialize the PCIe on next reset */
-				boot_strap |= 0x2;
-				regmap_writel(boot_strap, mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
-			} else {
-				regmap_writel(1, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-				/* Do not reinitialize the PCIe */
-				boot_strap &= ~0x2;
-				regmap_writel(boot_strap, mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
-				preboot_puts("NIC2_HBM configuration done, resetting...\n");
-				gaudi_reset_soc();
-			}
-			break;
-		case mmPSOC_PCI_PLL_BASE:
-			hl_pll_wa_clk_sw(&gaudi_pll_cfg[i]);
-			hl_uart_init();
-			break;
-		default:
-			break;
-		}
-	}
 }
diff --git a/gaudi/onix.c b/gaudi/onix.c
index 60969f0..f3e924b 100644
--- a/gaudi/onix.c
+++ b/gaudi/onix.c
@@ -47,5 +47,3 @@
 	/* baud rate doesn't support odd number */
 	return (factor + 1) & 0xFFFE;
 }
-
-void hl_pll_wa(void) {}
diff --git a/gaudi/pldm.c b/gaudi/pldm.c
index 044289c..c07682b 100644
--- a/gaudi/pldm.c
+++ b/gaudi/pldm.c
@@ -47,5 +47,3 @@
 	/* baud rate doesn't support odd number */
 	return (factor + 1) & 0xFFFE;
 }
-
-void hl_pll_wa(void) {}
diff --git a/goya/common.c b/goya/common.c
index adc5d7f..373209c 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -167,7 +167,6 @@
 	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
 }
 
-void hl_pll_wa(void) {}
 void preboot_sync_btl(void) {}
 
 int hl_pcie_init_only(void)
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 26153a8..8a49c58 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -137,7 +137,6 @@
 	return mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3;
 }
 
-void hl_pll_wa(void) {}
 void preboot_sync_btl(void) {}
 
 
